Forwarded-Upstream: Not needed
Author: James Henstridge
Source: http://code.djangoproject.com/changeset/12344

 [The] Python 2.6 package included unreleased fixes from the release26-maint
 branch. In particular it included the following bug fix to the Cookie module:

   http://bugs.python.org/issue5275

 When the django.test client builds up a WSGI environment, it passes a
 SimpleCookie? instance for HTTP_COOKIE. Prior to the change,
 django.http.parse_cookie() would copy the morsels directly. After the change,
 the values for all the cookies are the str() version of the morsels (e.g.
 "Set-Cookie: sessionid=value" instead of "value").

 This breaks any tests that depend on cookie handling (e.g. those that depend
 on cookie authentication).

                           <http://code.djangoproject.com/ticket/12720>

--- a/django/test/client.py (revision 11657)
+++ b/django/test/client.py (revision 12344)
@@ -194,5 +194,5 @@
         """
         environ = {
-            'HTTP_COOKIE':      self.cookies,
+            'HTTP_COOKIE':       self.cookies.output(header='', sep='; '),
             'PATH_INFO':         '/',
             'QUERY_STRING':      '',
